src/js/oncoprintlegendrenderer.js

```diff
@@ -148,14 +148,17 @@ var OncoprintLegendView = (function() {
 	view.$svg[0].setAttribute('width', everything_box.width);
 	view.$svg[0].setAttribute('height', everything_box.height);
     };
-    
+
     var ruleToSVGGroup = function(rule, view, model, target_svg, target_defs) {
 	var root = svgfactory.group(0,0);
 	var config = rule.getLegendConfig();
 	if (config.type === 'rule') {
 	    var concrete_shapes = rule.apply(config.target, model.getCellWidth(true), view.base_height);
+	    // generate backgrounds
+	    root.appendChild(svgfactory.bgrect(6, 20, 'rgb(190,190,190)'));
```

Comment:
confused.. this is not what we discussed

Reply:
You mean adding the gray backgrounds? I think it was discussed somewhere else and people agreed it was an improvement.

Reply:
not every rule will have a gray background, this is putting a very specific use case into a very general piece of code. what I had discussed with @rnugraha in the comment thread starting here https://github.com/cBioPortal/oncoprintjs/pull/55#discussion_r167663477 is adding a member to `Rule` that we can use for this purpose